### ISSUE TITLE
Cannot filter model years in query that uses comstock/resstock

### ISSUE THREAD

**Issue by daniel-thom** [dsgrid/dsgrid#212]

**Comment by lixiliu** [comment]
Yeah, I think w could split the filters into two types, one that can be applied to standalone datasets at the front and the other the end,  as the last query step before results. I think we can programmatically determine these filter types based on dimension records of standalone, project, and what is asked in the filter?

**Comment by daniel-thom** [comment]
We already have a way to specify post-filtering (https://github.com/dsgrid/dsgrid/blob/dt/apply-growth-rates/dsgrid/query/models.py#L402). It didn't work until yesterday.

I suppose we could reduce the filter specifications from two to one, and then figure out the optimal time to apply them. It does make sense. Many users just want the data filtered and don't care when it happens.

Previously, we talked about giving the user fine-grained control over when the filters happen. For the main issue discussed here, pre-filtering 1-2 model years would save a ton of work over post-filtering (40x reduction in data that has to be mapped to the project's dimensions). So, regardless of the point, I think we need to be able to pre-filter ResStock model years.

**Comment by daniel-thom** [comment]
My current opinion is that no changes are required for now.
- Most users should specify post-filters.
- For cases where a power user wants to specify a pre-filter they need to include all required dimension records for each dataset. In the example in this case, the query should have included 2018 in the pre-filter and then excluded it in the post-filter.

We could develop something more intelligent in the future.
